### PR TITLE
dw-dma: two enhancements and move to drivers/dw

### PR DESCRIPTION
**Pull request by lyakh**
Add a scatter capability, fix DONE bit and move to src/drivers/dw

### CODE REVIEW

**Comment by lyakh** [comment]
Build failed on HSW and BDW because DW_CTLL_D_SCAT_EN isn't defined on those platforms. I don't have DW DMA documents for those platforms, is that bit not present in their CTL_LO? Hm, I think I know what to do with it - DW_USE_HW_LLI isn't used on those platforms. I'll update the patches.

File: src/drivers/dw-dma.c [hunk 1]
```diff
@@ -88,6 +88,8 @@
 	(0x0040 + BYT_CHAN_OFFSET(chan))
 #define DW_CFG_HIGH(chan) \
 	(0x0044 + BYT_CHAN_OFFSET(chan))
+#define DW_DSR(chan) \
```
Comment (lgirdwood) [inline src/drivers/dw-dma.c]: This register may not be available on all versions of IP, I guess you have only tested on SUE ?

**Comment by lgirdwood** [comment]
@mengdonglin can someone validate this DMA update on BYT and APL

File: src/drivers/dw-dma.c (hunk 1 above)
Comment (lyakh) [reply src/drivers/dw-dma.c]: yes, but it is only used if the new .scatter flag is enabled, and so far it is only enabled on Sue Creek / by the SPI driver, so, at least this won't cause runtime problems. But if this register isn't available on some platforms, we *might* make it conditional on some preprocessor macro, that would make the code uglier though...

Comment (lgirdwood) [reply src/drivers/dw-dma.c]: SUE has the same IP as CNL, ICL for DW-DMA sou you could use the existing preprocessor macros.

**Comment by lyakh** [comment]
@lgirdwood what about this one? Have checks been performed? It has to be rebased yet again?

**Comment by lgirdwood** [comment]
@lyakh sorry, can you reabse for validation. @mengdonglin can someone validate pls before CNY.

**Comment by lgirdwood** [comment]
@lyakh can you rebase and I'll merge today.

**Comment by lyakh** [comment]
@lgirdwood interesting, after a "git mv" "git format-patch" produces a patch with just a rename, so, "git am" can apply it even if the file itself has changed. Apparently, this isn't working here, so, "git mv" gets conflicts for every change of the file.

**Comment by lgirdwood** [comment]
@zrombel can you check CI seems we have some server errors here.

**Comment by lyakh** [comment]
@lgirdwood can we merge now please?

**Pull request merged**